tests/e2e-pw/specs/shopper/shopper-checkout-purchase-site-editor.spec.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import test, { Page, expect } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import {
+	disableCardTestingProtection,
+	enableCardTestingProtection,
+} from '../../utils/devtools';
+import { getMerchant, getShopper } from '../../utils/helpers';
+import { activateTheme } from '../../utils/merchant';
+import { config } from '../../config/default';
+import {
+	addCartProduct,
+	confirmCardAuthentication,
+	emptyCart,
+	expectFraudPreventionToken,
+	fillCardDetails,
+	placeOrder,
+	setupCheckout,
+} from '../../utils/shopper';
+import { goToShop } from '../../utils/shopper-navigation';
+
+/**
+ * Tests for successful purchases with both card testing prevention enabled
+ * and disabled states using a site builder enabled theme.
+ */
+[ true, false ].forEach( ( cardTestingPreventionEnabled ) => {
+	test.describe( 'Successful purchase, site builder theme', () => {
+		let merchantPage: Page = null,
+			shopperPage: Page = null;
+		test.beforeAll( async ( { browser } ) => {
+			merchantPage = ( await getMerchant( browser ) ).merchantPage;
+			shopperPage = ( await getShopper( browser ) ).shopperPage;
+			await activateTheme( merchantPage, 'twentytwentyfour' );
+			if ( cardTestingPreventionEnabled ) {
+				await enableCardTestingProtection( merchantPage );
+			}
+		} );
+
+		test.afterAll( async () => {
+			await emptyCart( shopperPage );
+			await activateTheme( merchantPage, 'storefront' );
+			if ( cardTestingPreventionEnabled ) {
+				await disableCardTestingProtection( merchantPage );
+			}
+		} );
+
+		test.beforeEach( async () => {
+			await emptyCart( shopperPage );
+			await goToShop( shopperPage );
+			await addCartProduct( shopperPage );
+			await setupCheckout(
+				shopperPage,
+				config.addresses.customer.billing
+			);
+		} );
+
+		const sharedTestMethod = async (
+			page: Page,
+			card: any,
```

Comment:
Nitpick: we can replace `any` here with `typeof config.cards.basic`.

Reply:
Addressed in [61eed14](https://github.com/Automattic/woocommerce-payments/pull/10195/commits/61eed14c631dece9deb8fbe1edc522dae6cdcd01).